src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/GeneralizedCoastlineCheck.java

```diff
@@ -0,0 +1,111 @@
+package org.openstreetmap.atlas.checks.validation.linear.lines;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Segment;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.LineItem;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.tags.NaturalTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Flags generalized coastlines-- that is, OSM ways with the tag natural=coastline where x% or more
+ * node pairs are y or more meters apart. X and y can be specified in a {@link Configuration}.
+ * Coastlines can be represented in Atlas by {@link LineItem}s and may be members of
+ * {@link Relation}s.
+ *
+ * @author seancoulter
+ */
+public class GeneralizedCoastlineCheck extends BaseCheck<Long>
+{
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
```

Comment:
nit: could replace Arrays.asList with Collections.singletonList as we just have a single instruction.